driver-core/src/main/com/mongodb/internal/connection/ConnectionPool.java

```diff
@@ -37,6 +38,8 @@
 
     void invalidate();
 
+    void invalidate(ObjectId serviceId);
```

Comment:
LoadBalancedServer invalidates only a portion of the pool, identified by its serviceId